analytics/lib/counts.py

```diff
@@ -312,27 +311,41 @@ def do_increment_logging_stat(
         return
 
     table = stat.data_collector.output_table
+    id_args: Dict[str, Union[int, None]] = {}
+    conflict_args: List[str] = []
     if table == RealmCount:
         assert isinstance(model_object_for_bucket, Realm)
-        id_args: Dict[
-            str, Optional[Union[Realm, UserProfile, Stream, "RemoteRealm", "RemoteZulipServer"]]
-        ] = {"realm": model_object_for_bucket}
+        id_args = {"realm_id": model_object_for_bucket.id}
+        conflict_args = ["realm_id"]
     elif table == UserCount:
         assert isinstance(model_object_for_bucket, UserProfile)
-        id_args = {"realm": model_object_for_bucket.realm, "user": model_object_for_bucket}
+        id_args = {
+            "realm_id": model_object_for_bucket.realm_id,
+            "user_id": model_object_for_bucket.id,
+        }
+        conflict_args = ["user_id"]
     elif table == StreamCount:
         assert isinstance(model_object_for_bucket, Stream)
-        id_args = {"realm": model_object_for_bucket.realm, "stream": model_object_for_bucket}
+        id_args = {
+            "realm_id": model_object_for_bucket.realm_id,
+            "stream_id": model_object_for_bucket.id,
+        }
+        conflict_args = ["stream_id"]
     elif table == RemoteInstallationCount:
         assert isinstance(model_object_for_bucket, RemoteZulipServer)
-        id_args = {"server": model_object_for_bucket, "remote_id": None}
+        id_args = {"server_id": model_object_for_bucket.id, "remote_id": None}
+        conflict_args = ["server_id"]
     elif table == RemoteRealmCount:
         assert isinstance(model_object_for_bucket, RemoteRealm)
         id_args = {
-            "server": model_object_for_bucket.server,
-            "remote_realm": model_object_for_bucket,
+            "server_id": model_object_for_bucket.server_id,
+            "remote_realm_id": model_object_for_bucket.id,
             "remote_id": None,
         }
+        conflict_args = [
+            "server_id",
+            "realm_id",
```

Comment:
Is this supposed to be `remote_realm_id`?

Reply:
Unique Constraints for RemoteRealmCount are set to following
```py
UniqueConstraint(
    fields=["server", "realm_id", "property", "subgroup", "end_time"],
    condition=Q(subgroup__isnull=False),
    name="unique_remote_realm_installation_count",
),
UniqueConstraint(
    fields=["server", "realm_id", "property", "end_time"],
    condition=Q(subgroup__isnull=True),
    name="unique_remote_realm_installation_count_null_subgroup",
),
UniqueConstraint(
    fields=["server", "remote_id"],
    # As with RemoteInstallationCount above, remote_id may
    # be null; since null != null, this condition is not
    # strictly necessary, but serves to make the property
    # more explicit.
    condition=Q(remote_id__isnull=False),
    name="unique_remote_realm_installation_count_server_id_remote_id",
),
```
if we added `remote_realm_id` to on conflict keys, we will get an Unique Constraints Not set error.